src/extension.ts

```diff
@@ -597,6 +598,17 @@ export class Extension implements RunHooks {
       await installBrowsers(this._vscode, model);
   }
 
+  private async _markTestRunFailed(testRun: vscodeTypes.TestRun, testItem: vscodeTypes.TestItem, result: reporterTypes.TestResult) {
+    let aiContext: string | undefined;
+    const snapshot = result.attachments.find(a => a.name === 'pageSnapshot');
+    if (snapshot && snapshot.path) {
+      const contents = await this._vscode.workspace.fs.readFile(this._vscode.Uri.file(snapshot.path));
+      aiContext = `### Page Snapshot at Failure\n${contents.toString()}`; // cannot use ``` codeblocks, vscode markdown does not support it
```

Comment:
Don't you need two `\n` to make a paragraph?

Reply:
some renderers do, you're right. vscode didn't require it for some reason, but it still makes sense to make it two newlines in case vscode markdown rendering ever changes.